File: kvt-projekat2024/src/app/models/Rate.ts
import { Review } from "./Review";

export class Rate {
  id: number;
  equipment: number;
  staff: number;
  hygiene: number;
  space: number;
  reviews: Review[];

  constructor(id: number, equipment: number, staff: number, hygiene: number, space: number, reviews: Review[]) {
    this.id = id;
    this.equipment = equipment;
    this.staff = staff;
    this.hygiene = hygiene;
    this.space = space;
    this.reviews = reviews;
  }
}

File: src/main/java/rs/ac/uns/ftn/svt/model/Rate.java
package rs.ac.uns.ftn.svt.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
public class Rate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer equipment;

    private Integer staff;

    private Integer hygiene;

    private Integer space;

    @OneToMany(mappedBy = "rate")
    private List<Review> reviews;

}
